### PR TITLE
fix: align is visible with snapshot semantics

### LINKED ISSUE #271: Align `is visible` semantics with snapshot presence checks
## Summary
Align `is visible` with the practical presence semantics teams already rely on from `wait text`.

## Current behavior
`is visible` can fail for list items that are plainly present in the snapshot, while `wait text` succeeds for the same content.

## Proposed scope
- Review the visibility predicate used for selector-based assertions.
- Align the behavior with actual snapshot presence and viewport visibility expectations.
- Add tests that cover list items and off-screen/zero-rect cases.

## Acceptance criteria
- Elements present in the visible snapshot do not fail `is visible` for purely internal hittability reasons.
- `is visible`, `is exists`, and `wait text` have clearly documented semantics.

### PR DESCRIPTION
**Pull request by thymikee**
## Summary

Closes #271.

Align `is visible` with viewport-aware snapshot presence semantics so visible list content does not fail for internal hittability reasons while off-screen and zero-rect cases still fail.
Clarify `is visible`, `is exists`, and `wait text` semantics in docs and skill guidance. Touched files: 6. Scope expanded beyond the initial command family: yes, for tests and docs/skills only; code changes remain in the `is` assertion path.

## Validation

pnpm format
pnpm typecheck
pnpm test:unit
pnpm test:smoke

### CODE REVIEW

File: src/daemon/is-predicates.ts [hunk 1]
```diff
@@ -39,9 +44,88 @@ export function evaluateIsPredicate(params: {
...
+function resolveVisibilityAnchor(
+  node: SnapshotState['nodes'][number],
+  nodes: SnapshotState['nodes'] | undefined,
+): SnapshotState['nodes'][number] | null {
+  if (!nodes?.length) return null;
+  let current = node;
+  const visited = new Set<number>();
+  while (typeof current.parentIndex === 'number' && !visited.has(current.index)) {
+    visited.add(current.index);
+    const parent = nodes[current.parentIndex];
```
[the remark this answers is not in the document]
Comment (thymikee) [reply src/daemon/is-predicates.ts]: Addressed in 4b768f44. `evaluateIsPredicate` now requires `nodes`, ancestor lookup follows `parentIndex` through a `Map` keyed by `node.index`, and there is a regression test covering sparse/pruned node indexes.

**Review by thymikee — COMMENTED** [review]
## Re-review

All three items addressed:
1. `nodes` is now required in `evaluateIsPredicate`
2. `isUsefulVisibilityAnchor` has the explanatory comment about scroll containers reporting full content frame
3. Docs double negative fixed — reads clearly now

One minor note: `isAssertionVisible` is only used for `visible`/`hidden` predicates, while `text` still uses the old `isNodeVisible`. The diff makes this intentional (text predicate only needs node existence, not viewport awareness) — worth confirming that's the desired long-term semantic for `is text`.

LGTM.
